packages/pcd/gpc-pcd/src/GPCPCDPackage.ts

```diff
@@ -62,13 +65,42 @@ async function checkProofArgs(args: GPCPCDArgs): Promise<{
   }
   const proofConfig = deserializeGPCProofConfig(args.proofConfig.value);
 
-  if (!args.pod.value) {
+  // Filter out PODPCD records.
+  const podRecord = Object.fromEntries(
+    Object.entries(args).filter(([argName, _arg]) =>
+      argName.startsWith(`${PODPCD_ARG_PREFIX}_`)
+    )
+  ) as PODPCDRecordArg;
+
+  if (Object.keys(podRecord).length === 0) {
     throw new Error("No PODPCD value provided");
   }
-  const podPCD = await PODPCDPackage.deserialize(args.pod.value.pcd);
-  if (!isPODPCD(podPCD)) {
-    throw new Error("Wrong PCD type provided for PODPCD");
-  }
+
+  const pods = Object.fromEntries(
+    await Promise.all(
+      Object.entries(podRecord).map(async ([prefixedPODName, podPCDArg]) => {
+        const podNameMatch = prefixedPODName.match(PODPCD_ARG_REGEXP);
+
+        if (podNameMatch === null) {
+          throw new Error(`Invalid PODPCD argument name ${prefixedPODName}`);
+        }
+
+        // Strip away prefix
+        const podName = podNameMatch[2];
+
+        if (!podPCDArg.value) {
+          throw new Error(`No PODPCD value provided for POD ${podName}`);
+        }
+
+        const podPCD = await PODPCDPackage.deserialize(podPCDArg.value.pcd);
```

Comment:
this may or may not throw depending on user input and deserialization logic

Reply:
I think that's okay.  prove() should fail if we're given an invalid PCD.  Would you suggest something else?

Reply:
One consideration I'm thinking about is UX. For example, I wonder what is shown on the prove screen given an exception at this location. Errors bubble up to either a Developer, User, or Rollbar, so I generally like to think about how they will manifest to one of those audiences. For example, when an error occurs during the loading of a Podbox pipeline, it's recorded into an array of errors that then get saved for viewing later. Just expanding on my thought process here.